dora/core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java

```diff
@@ -379,6 +380,18 @@ public void rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options)
 
       mDoraClient.rename(srcUfsFullPath.toString(), dstUfsFullPath.toString(), mergedOptions);
     } catch (RuntimeException ex) {
+      if (ex instanceof StatusRuntimeException) {
+        Status.Code code = ((StatusRuntimeException) ex).getStatus().getCode();
+        if (code == Status.FAILED_PRECONDITION.getCode()) {
+          // throw new RuntimeException(ex.getMessage());
```

Comment:
just not sure this is the appropriate exception to throw

Reply:
@Jackson-Wang-7 WDYT?

Reply:
for the FAILED_PRECONDITION, I think it may be not allowed to rename the file to file but it's ok to rename the file into a directory. so we may need a log here and wait for the result of 2nd try?